components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/context/SessionContext.java

```diff
@@ -56,6 +54,16 @@ public void setAuthenticatedIdPs(Map<String, AuthenticatedIdPData> authenticated
         this.authenticatedIdPs = authenticatedIdPs;
     }
 
+    public void setAuthenticatedIdPsPerApp(Map<String, Map<String, AuthenticatedIdPData>> authenticatedIdPsPerApp) {
```

Comment:
Serial version UID of the the class will need to be updated as this is modifying a serializable class.
Good to add a note on the impact on the existing session context data reading.

Reply:
If we introduce a new uuid, it may break the existing sessions. Since it is not a major change, I think we can skip it. I handled this change gracefully.